### PR TITLE
add token address

### PR DESCRIPTION
**Pull request by neoiss**
**NOTE**

#### Please enable "Allow edits by maintainers" while putting up the PR.

---

> - If you would like to add a `volume` adapter please submit the PR [here](https://github.com/DefiLlama/adapters).
> - If you would like to add a `liquidations` adapter, please refer to [this readme document](https://github.com/DefiLlama/DefiLlama-Adapters/tree/main/liquidations) for details.

1. Once your adapter has been merged, it takes time to show on the UI. If more than 24 hours have passed, please let us know in Discord.
2. Sorry, We no longer accept fetch adapter for new projects, we prefer the tvl to computed from blockchain data, if you have trouble with creating a the adapter, please hop onto our discord, we are happy to assist you.
3. Please fill the form below  **only if the PR is for listing a new protocol** else it can be ignored/replaced with reason/details about the PR
4. **For updating listing info** It is a different repo, you can find your listing in this file: https://github.com/DefiLlama/defillama-server/blob/master/defi/src/protocols/data2.ts, you can  edit it there and put up a PR
5. Do not edit/push `package-lock.json` file as part of your changes, we use lockfileVersion 2, and most use v1 and using that messes up our CI
6. No need to go to our discord and announce that you've created a PR, we monitor all PRs and will review it asap

---

### CODE REVIEW

**Comment by llamatester** [comment]
Error while running adapter at projects/butternetwork: 
        
 
  
    
    
    Error: 
        Unknown chain(s): optimistic-ethereum, ainn
        Note: if you think that the chain is correct but missing from our list, please add it to 'projects/helper/chains.json' file
        
    Error saving cache: Error: ENOENT: no such file or directory, open '/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/local_cache/zlib-1.0/BlockCache.json'
        at Object.openSync (node:fs:596:3)
        at Object.writeFileSync (node:fs:2322:35)
        at process.saveCache (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/cache.js:152:17)
        at process.emit (node:events:529:35)
        at process.exit (node:internal/process/per_thread:191:15)
        at process.handleError (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/utils/handleError.js:18:11)
        at process.emit (node:events:529:35)
        at emit (node:internal/process/promises:149:20)
        at processPromiseRejections (node:internal/process/promises:283:27)
        at process.processTicksAndRejections (node:internal/process/task_queues:96:32) {
      errno: -2,
      syscall: 'open',
      code: 'ENOENT',
      path: '/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/local_cache/zlib-1.0/BlockCache.json'
    }

**Comment by llamatester** [comment]
Error while running adapter at projects/butternetwork: 
        
 
  
    
    
    Error: Multicall failed! 
     [chain: zksync] [fail count: 3] [abi: function balanceOf(address account) view returns (uint256)]  
        target: 0x493257fd37edb34451f62edf8d2a0c418852ba4c params: 0xfeb2b97e4efce787c08086dc16ab69e063911380]
        target: 0x3355df6d4c9c3035724fd0e3914de96a5a83aaf4 params: 0xfeb2b97e4efce787c0...
    Error saving cache: Error: ENOENT: no such file or directory, open '/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/local_cache/zlib-1.0/BlockCache.json'
        at Object.openSync (node:fs:596:3)
        at Object.writeFileSync (node:fs:2322:35)
        at process.saveCache (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/cache.js:152:17)
        at process.emit (node:events:529:35)
        at process.exit (node:internal/process/per_thread:191:15)
        at process.handleError (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/utils/handleError.js:18:11)
        at process.emit (node:events:529:35)
        at emit (node:internal/process/promises:149:20)
        at processPromiseRejections (node:internal/process/promises:283:27)
        at process.processTicksAndRejections (node:internal/process/task_queues:96:32) {
      errno: -2,
      syscall: 'open',
      code: 'ENOENT',
      path: '/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/local_cache/zlib-1.0/BlockCache.json'
    }

**Comment by llamatester** [comment]
Error while running adapter at projects/butternetwork: 
        
 
  
    
    
    Error: Multicall failed! 
     [chain: zksync] [fail count: 3] [abi: function balanceOf(address account) view returns (uint256)]  
        target: 0x493257fd37edb34451f62edf8d2a0c418852ba4c params: 0xbef06a32166c4b819ff04ccfa887733b8bb67eb5]
        target: 0x3355df6d4c9c3035724fd0e3914de96a5a83aaf4 params: 0xbef06a32166c4b819f...
    Error saving cache: Error: ENOENT: no such file or directory, open '/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/local_cache/zlib-1.0/BlockCache.json'
        at Object.openSync (node:fs:596:3)
        at Object.writeFileSync (node:fs:2322:35)
        at process.saveCache (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/cache.js:152:17)
        at process.emit (node:events:529:35)
        at process.exit (node:internal/process/per_thread:191:15)
        at process.handleError (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/utils/handleError.js:18:11)
        at process.emit (node:events:529:35)
        at emit (node:internal/process/promises:149:20)
        at processPromiseRejections (node:internal/process/promises:283:27)
        at process.processTicksAndRejections (node:internal/process/task_queues:96:32) {
      errno: -2,
      syscall: 'open',
      code: 'ENOENT',
      path: '/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/local_cache/zlib-1.0/BlockCache.json'
    }

File: projects/butternetwork/index.js [hunk 1]
```diff
@@ -140,8 +140,76 @@ const config = {
...
+    mosContract: "0xfeB2b97e4Efce787c08086dC16Ab69E063911380",
+    tokens: {
+      USDT: ADDRESSES.optimism.USDT,
+      USDC: ADDRESSES.optimism.USDC,
+      WBTC: ADDRESSES.optimism.WBTC,
+      WETH: ADDRESSES.optimism.WETH_1,
+      DAI: ADDRESSES.optimism.DAI,
+    }
+  },
+  zksync: {
```
Comment (g1nt0ki) [inline projects/butternetwork/index.js:167]: change this to era

**Comment by llamatester** [comment]
The adapter at projects/butternetwork exports TVL: 
        
 
 
    merlin                    6.24 M
    klaytn                    302.42 k
    bsc                       121.40 k
    ethereum                  119.22 k
    optimism                  98.03 k
    arbitrum                  88.29 k
    era                       75.00 k
    polygon                   73.44 k
    conflux                   51.12 k
    tron                      39.26 k
    base                      29.81 k
    linea                     18.85 k
    blast                     13.35 k
    scroll                    11.61 k
    mantle                    10.10 k
    near                      179
    map                       0
    
    total                    7.29 M

**Comment by g1nt0ki** [comment]
@neoiss thanks for the PR

**Pull request merged**